handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -2078,7 +2076,12 @@ private void flush(ChannelHandlerContext ctx, ChannelPromise promise) throws Exc
     public void handlerAdded(final ChannelHandlerContext ctx) throws Exception {
         this.ctx = ctx;
         Channel channel = ctx.channel();
-        pendingUnencryptedWrites = new SslHandlerCoalescingBufferQueue(channel, 16);
+        pendingUnencryptedWrites = new SslHandlerCoalescingBufferQueue(channel, 16, engineType.wantsDirectBuffer) {
+            @Override
+            protected int wrapDataSize() {
+                return SslHandler.this.wrapDataSize;
```

Comment:
How about we move the `wrapDataSize` field into the `SslHandlerCoalescingBufferQueue` and make the class final, instead?

Reply:
Sure we can do this... Let me take care

Reply:
actually this not really works because there is also a setWrapDataSize(...) method in `SslHandler`.

Reply:
can't that delegate to a setter on the coalescing buffer queue?

Reply:
The method might be called before the queue is created